library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -393,6 +393,10 @@ Http::FilterHeadersStatus PlatformBridgeFilter::encodeHeaders(Http::ResponseHead
   // error callback (rather than an HTTP response).
   const auto error_code_header = headers.get(Http::InternalHeaders::get().ErrorCode);
   if (!error_code_header.empty()) {
+    // This must be set here, since we won't be delegating to FilterBase.
+    response_filter_base_->state_.stream_complete_ = end_stream;
```

Comment:
IMO it would be cleaner to set this universally at the beginning of the function, so you could either allow it to be reset in the common function below or have some type of preamble function that is split out?

Reply:
Updated. MK what you think.